### PR TITLE
fix http client idle timeout to apply only to idleness in the pool

### PR DESCRIPTION
**Pull request by johnrengelman**
This change is [Reviewable](https://reviewable.io/reviews/ratpack/ratpack/1470)

### CODE REVIEW

File: ratpack-core/src/test/groovy/ratpack/http/client/HttpClientIdleTimeoutSpec.groovy [hunk 1]
```diff
@@ -116,6 +127,7 @@ class HttpClientIdleTimeoutSpec extends BaseHttpClientSpec {
 
     @Override
     void error(Context context, Throwable throwable) throws Exception {
+      throwable.printStackTrace()
```
Comment (drmaas) [inline ratpack-core/src/test/groovy/ratpack/http/client/HttpClientIdleTimeoutSpec.groovy]: remove

Comment (johnrengelman) [reply ratpack-core/src/test/groovy/ratpack/http/client/HttpClientIdleTimeoutSpec.groovy]: Whoops.

Comment (drmaas) [reply ratpack-core/src/test/groovy/ratpack/http/client/HttpClientIdleTimeoutSpec.groovy]: @johnrengelman if you add another test where you sleep > timeout in between requests, you can assert that the channel is closed/null? and not in the pool.

Comment (johnrengelman) [reply ratpack-core/src/test/groovy/ratpack/http/client/HttpClientIdleTimeoutSpec.groovy]: That's what the 2nd test case in `test client with idle timeout exceeded()`  tests. the `reused=false` condition on that test.
